src/Game.js

```diff
@@ -0,0 +1,78 @@
+function Game() {
+  this.frames = [[[]]];
+  this.currentFrame = 0;
+}
+
+Game.prototype.addScore = function(score) {
+  if (this.currentFrame === 9 && this.frames[this.currentFrame].length >= 1) {
+    this._tenthFrame(score);
+  } else {
+    this._checkFrame();
+    this._currentFrameScore().push(score);
+    this._applyBonus(score);
+    this._checkBonus(score);
+  }
+};
+
+Game.prototype.frameTotal = function() {
+ return this._currentFrameScore().reduce(function(a, b) {
+    return a + b;
+  }, 0);
+};
+
+Game.prototype.total = function() {
+  var total = 0;
+  for (var i = 0; i < this.frames.length; i++) {
+    total += this.frames[i][0].reduce(function(a, b) {
+      return a + b;
+    }, 0);
+  }
+  return total;
+};
+
+Game.prototype._tenthFrame = function(score) {
+  if ((this._currentFrameBonus() > 0 || this._currentFrameScore().length == 1) && (this._currentFrameScore().length < 3)) {
+    this._currentFrameScore().push(score);
+    this._checkBonus(score);
+    if (this.frames[8][1] == 1) {
+      this.frames[8][0].push(score);
+      this.frames[this.currentFrame - 1][1] -= 1;
+    }
+  }
+};
+
+Game.prototype._checkFrame = function() {
+  if (this._currentFrameScore().length == 2 || this._currentFrameScore()[0] == 10) {
+    this._addFrame();
+  }
+};
+
+Game.prototype._addFrame = function() {
+  this.currentFrame += 1;
+  this.frames.push([[]]);
+};
+
+Game.prototype._checkBonus = function(score) {
```

Comment:
It feels like `_checkBonus` is going to return a boolean, but it actually changes some data.  Could it be renamed?